File: Backend/Hotels/src/main/java/com/example/Hotels/Domain/Hotel.java
package com.example.Hotels.Domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Document
public class Hotel {
    @Id
    private String registrationId;
    private String hotelName;
    private String ownerName;
    private String city;
    private double rating;
    private byte[] image;
    private List<Menu> menus;
    private List<Room> rooms;
}

File: Frontend/Data0/src/app/Domain/General-Domain/Hotel.ts
import { Menu } from "./Menu";
import { Room } from "./Room";

export class Hotel{
    registrationId: string;
    hotelName: string;
    ownerName: string;
    city: string;
    rating: number;
    image: string;
    menus: Menu[];
    rooms: Room[];
    constructor(){
        this.registrationId="",
        this.hotelName="",
        this.ownerName="",
        this.city="",
        this.rating=0,
        this.image="",
        this.menus=[],
        this.rooms=[]
    }
}